test/dlc_tests/sanity/test_scan_vulnerability_class_functionality.py

```diff
@@ -0,0 +1,176 @@
+    assert (
+        vuln_list_two_minus_one.vulnerability_list == two_minus_one_saved_results
+    ), f"{vuln_list_two_minus_one.vulnerability_list} \n does not match \n {two_minus_one_saved_results}"
+    assert (ecr_image_vulnerability_list1 - ecr_image_vulnerability_list1) == None, "Failed"
+
+
+@pytest.mark.usefixtures("sagemaker")
+@pytest.mark.model("N/A")
+@pytest.mark.integration("Check if Allowlist files are read successfully")
+def test_allowlist_file_read():
+    minimum_sev_threshold = "HIGH"
+    ecr_image_vulnerability_list = ECREnhancedScanVulnerabilityList(minimum_severity=CVESeverity[minimum_sev_threshold])
+    ecr_image_vulnerability_list.construct_allowlist_from_file("./sanity/resources/allowlist_to_read.json")
+
+    with open("./sanity/resources/allowlist1.json", "r") as f:
+        target_allowlist = json.load(f)
+
+    assert (
+        ecr_image_vulnerability_list.vulnerability_list == target_allowlist
+    ), f"{ecr_image_vulnerability_list.vulnerability_list} \n does not match \n {target_allowlist} "
+
+
+@pytest.mark.usefixtures("sagemaker")
+@pytest.mark.model("N/A")
+@pytest.mark.integration("Check if cmp operator works properly")
+def test_compare_operator():
+    """
+    This method tests the compare i.e. __cmp__ functionality of ECREnhancedScanVulnerabilityList class.
+    It reads ecr_scan_result1.json file in TWO different objects - object1a and object1b - and checks if
+    __cmp__ returns True when applied on the 2 objects. It also reads ecr_scan_result2.json as object2 and 
+    then compares object2 with object1a and expects False as an outcome.
+    """
+    minimum_sev_threshold = "HIGH"
+
+    with open("./sanity/resources/ecr_scan_result1.json", "r") as f:
+        scan_results = json.load(f)
+    ecr_image_vulnerability_list1a = ECREnhancedScanVulnerabilityList(
+        minimum_severity=CVESeverity[minimum_sev_threshold]
+    )
+    ecr_image_vulnerability_list1a.construct_allowlist_from_ecr_scan_result(scan_results)
+
+    with open("./sanity/resources/ecr_scan_result1.json", "r") as f:
+        scan_results = json.load(f)
+    ecr_image_vulnerability_list1b = ECREnhancedScanVulnerabilityList(
+        minimum_severity=CVESeverity[minimum_sev_threshold]
+    )
+    ecr_image_vulnerability_list1b.construct_allowlist_from_ecr_scan_result(scan_results)
+
+    with open("./sanity/resources/ecr_scan_result2.json", "r") as f:
+        scan_results = json.load(f)
+    ecr_image_vulnerability_list2 = ECREnhancedScanVulnerabilityList(
+        minimum_severity=CVESeverity[minimum_sev_threshold]
+    )
+    ecr_image_vulnerability_list2.construct_allowlist_from_ecr_scan_result(scan_results)
+
+    assert ecr_image_vulnerability_list1a.__cmp__(
+        ecr_image_vulnerability_list1b
+    ), f"Same vulnerability lists not marked same"
+    assert not ecr_image_vulnerability_list1a.__cmp__(
+        ecr_image_vulnerability_list2
+    ), f"Different vulnerability lists marked same"
```

Comment:
This is mostly redundant since `__cmp__` is not used in python3 anymore. Having rich comparison operators, specifically `__eq__` might prove to be beneficial in the future.

Reply:
Added `__eq__` and `__ne__`. `__cmp__` still remains for backward compatibility. Will remove in future.